de/nmichael/efa/data/efacloud/TableBuilder.java

```diff
@@ -0,0 +1,520 @@
+package de.nmichael.efa.data.efacloud;
+
+import de.nmichael.efa.core.Backup;
+import de.nmichael.efa.data.storage.EfaCloudStorage;
+import de.nmichael.efa.data.storage.MetaData;
+import de.nmichael.efa.data.storage.StorageObject;
+import de.nmichael.efa.gui.EfaBaseFrame;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+
+public class TableBuilder {
+
+    /**
+     * There are essentially two namespaces for efa2 records the efa2 record type name like
+     * efa2persons and the record's table class name, like Persons. This here maps the name spaces
+     */
+    public static final HashMap<String, String> efaCloudTableNames = new HashMap<>();
```

Comment:
I'll have a look into it. When I started exercise I was not clear about the different ways a record is referenced and created Hashmaps to be able to refer whatever was needed. Also an exercise for cleansing the design, You're right.

Reply:
I finally looked into it. The reason is simple: we do have a local XML data base where records have a StorageObject to which they are stored, e.g. de.nmichael.efa.core.config.EfaTypes. The record definition is in de.nmichael.efa.core.config.EfaTypeRecord and to reference the StorageObject there is are getter & setter in IDataAccess (set/getStorageType()), e.g. 'efa2types'. Now the SQL DB of the efaCloud server uses tables to store the record. Their names are needed for the client-server communication. Now the question is, why I did not use the "storageType" namespace. Well, I just didn't understand what best to use when building the server tables. I will change it now to the StorageType.

Reply:
I tried to remove the mapping and just used the plain storage type names as table names, but this resulted in some ArrayOutOfBoundsExceptions in the synchronization. They seemed to be coming from the server... but I haven't debugged it more. I'm leaving it as is for now. But it's maybe something that should be changed before this feature becomes public as it would affect the table names on the server.

Reply:
I am going to do that change today (it is the morning of 4th Oct here already). It does affect the server so it must be done on both sides to work. Please go ahead and merge your code changes. I will then join the git type of collaboration as proposed.